src/System.Private.Uri/src/System/Uri.cs

```diff
@@ -2108,13 +2108,21 @@ private unsafe ParsingError PrivateParseMinimal()
                             // Only FILE scheme may have UNC Path flag set
                             _flags |= Flags.UncPath;
                             idx = i;
+
+                            _syntax = IsWindowsSystem ? _syntax : UriParser.UnixFileUri;
```

Comment:
This codepath is hit only in cases of where the unc uri is with file://, for cases of \\ this codepath is not hit. I had to change the iri test case because the test was doing file://\\unc\<rest_of_path> concatenation. Passing a value like \\host\path1\path2, will not be affected by this change.